app/assets/js/generator.js

```diff
@@ -1,76 +1,113 @@
+/*global instantsearch, timeago */
 (function (win) {
   'use strict';
 
   var $ = win.jQuery;
-
-  // Settings for doT.js
   var doT = win.doT;
   doT.templateSettings.interpolate = /<\%=([\s\S]+?)\%\>/g;
-  doT.templateSettings.conditional = /<\%if( else)?\s*([\s\S]*?)\s*\%>/g;
-  doT.templateSettings.iterate = /<\%each\s*(?:\%>|([\s\S]+?)\s*\:\s*([\w$]+)\s*(?:\:\s*([\w$]+))?\s*\%>)/g;
+  var itemTemplate = doT.template($('#template-item').text());
+
+  function highlight(item, key) {
+    if (item._highlightResult[key]) {
+      return item._highlightResult[key].value;
+    }
+    return item[key];
+  }
+
+  function getUrl(item) {
+    if (item.homepage) {
+      return item.homepage;
+    }
+    if (item.githubRepo) {
+      return 'https://github.com/' + item.githubRepo.user + '/' + item.githubRepo.project;
+    }
+    return 'https://npmjs.com/package/' + item.name;
+  }
+
+  function getDownloadClass(downloads) {
+    if (downloads < 1000) {
+      return null;
+    }
+    if (downloads < 10000) {
+      return 'generator--downloads-t1';
+    }
+    if (downloads < 50000) {
+      return 'generator--downloads-t2';
+    }
+    if (downloads < 100000) {
+      return 'generator--downloads-t3';
+    }
+    return 'generator--downloads-t4';
+  }
+
 
   $(function() {
+
     var instructions = $('#instructions');
     $('#instructions-toggle').on('click', function (e) {
       e.preventDefault();
       instructions.toggle('fast');
     });
 
-    var tpl = doT.template($('#plugins-all-template').text());
-    var pluginsAll = $('#plugins-all');
-    var lastSha = win.Cookies.get('sha');
+    var search = instantsearch({
+      appId: 'OFCNCOG2CU',
+      apiKey: '14ddf54fd4f3435c1cd4038395a0cf10',
+      indexName: 'npm-search',
+      searchParameters: {
+        filters: 'deprecated:false AND keywords:yeoman-generator'
```

Comment:
`filters: 'deprecated:false AND computedKeywords:yeoman-generator'`

I changed the Yarn index to now have a subset of the complete registry, they need the keyword _and_ start with `generator-`. You can see (and edit if necessary) the code for that [here](https://github.com/algolia/npm-search/blob/master/src/formatPkg.js#L217-L221)

Don't hesitate to reach out if there's any questions about this